src/vs/workbench/electron-sandbox/parts/titlebar/titlebarPart.ts

```diff
@@ -220,26 +199,8 @@ export class TitlebarPart extends BrowserTitleBarPart {
 		return ret;
 	}
 
-	override updateLayout(dimension: Dimension): void {
-		this.lastLayoutDimensions = dimension;
-
-		if (getTitleBarStyle(this.configurationService) === 'custom') {
-			if (isMacintosh || this.currentMenubarVisibility === 'hidden') {
-				this.rootContainer.style.height = `${100.0 * getZoomFactor()}%`;
-				this.rootContainer.style.width = `${100.0 * getZoomFactor()}%`;
-				this.rootContainer.style.transform = `scale(${1 / getZoomFactor()})`;
-			} else {
-				this.rootContainer.style.height = `100%`;
-				this.rootContainer.style.width = `100%`;
-				this.rootContainer.style.transform = '';
-			}
-
-			runAtThisOrScheduleAtNextAnimationFrame(() => this.adjustTitleMarginToCenter());
-
-			if (this.customMenubar) {
-				const menubarDimension = new Dimension(0, dimension.height);
-				this.customMenubar.layout(menubarDimension);
-			}
-		}
+	override updateStyles(): void {
+		super.updateStyles();
+		this.nativeHostService.updateTitleBarOverlay(this.element.style.backgroundColor, this.element.style.color);
```

Comment:
Could we be more clever here and only go to main process when we know something changed? I am a little bit worried that `updateStyles` might be called a lot more often then needed.